ksqldb-engine/src/test/java/io/confluent/ksql/structured/SchemaKSourceFactoryTest.java

```diff
@@ -336,7 +339,27 @@ public void shouldCreateNonWindowedTableSourceV2WithNewPseudoColumnVersionIfNoOl
     );
 
     // Then:
-    assertThat(((TableSource) result.getSourceStep()).getPseudoColumnVersion(), equalTo(NEW_PSEUDOCOLUMN_VERSION));
+    assertThat(((TableSource) result.getSourceStep()).getPseudoColumnVersion(), equalTo(SystemColumns.CURRENT_PSEUDOCOLUMN_VERSION_NUMBER));
+    assertValidSchema(result);
+  }
+
+  //this state shouldn't exist in normal use, testing for completeness
+  @Test
+  public void shouldReplaceOldWTableSourceV2WithMatchingPseudoColumnVersion() {
```

Comment:
What is an "OldWTableSourceV2"? See other discussion about the meaning of "old" in these tests.